### PR TITLE
Add fast beam search emformer

### PR DESCRIPTION
**Pull request by ezerhouni**
This PR adds ```fast_beam_search_one_best``` to the following models :

- [X] pruned_stateless_emformer_rnnt2

- [x] conv_emformer_transducer_stateless

### CODE REVIEW

File: sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py
Comment (csukuangfj) [inline sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py:19]: We are using [pep8](https://peps.python.org/pep-0008/#imports) style here:

```
Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.

You should put a blank line between each group of imports.
```

And we use [isore](https://github.com/PyCQA/isort) to sort imports automatically

File: sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py
Comment (csukuangfj) [inline sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py:164]: `assert False` throws an `AssertionError` exception.
I suggest
```python3
raise ValueError(f''Decoding method {decoding_method} is not supported.")
```
It expresses the intention directly.

File: sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py
Comment (csukuangfj) [inline sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py:204]: Please refactor it in a way that makes it easier to support other decoding methods.

File: sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py
Comment (csukuangfj) [inline sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py:177]: I suggest refactoring it into two functions:
- run_model_and_do_greedy_search
- run_model_and_do_fast_beam_search

It makes the code more readable and more maintainable, I think.

If we want to support another decoding method, we can just create a new function to do that.

----

It is just a suggestion. You can leave it as it is.

File: sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py
Comment (ezerhouni) [reply sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py:19]: Usually the pre-commit hook was taking care of it, that's why I notice that there are none. Will fix it

File: sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py
Comment (ezerhouni) [reply sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py:177]: It make sense, I will finish first the work of conv_emformer at first

File: sherpa/bin/conv_emformer_transducer_stateless/decode.py
Comment (csukuangfj) [inline sherpa/bin/conv_emformer_transducer_stateless/decode.py:2]: Please keep the authors info.

File: sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py
Comment (ezerhouni) [reply sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py:20]: It should be fine by now

File: sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py
Comment (ezerhouni) [reply sherpa/bin/pruned_stateless_emformer_rnnt2/decode.py]: Done

**Comment by csukuangfj** [comment]
Could you also update GitHub CI for https://github.com/k2-fsa/sherpa/blob/master/.github/workflows/run-streaming-conv-emformer-test.yaml#L36

to include
https://github.com/k2-fsa/sherpa/blob/fb0ee4adfdd5e490111ada244e0713410561d2dd/.github/workflows/run-streaming-conformer-test.yaml#L37

https://github.com/k2-fsa/sherpa/blob/fb0ee4adfdd5e490111ada244e0713410561d2dd/.github/workflows/run-streaming-conformer-test.yaml#L127

You can apply a label to this PR with the name `ready` to trigger GitHub actions.

File: sherpa/bin/conv_emformer_transducer_stateless/decode.py
Comment (ezerhouni) [reply sherpa/bin/conv_emformer_transducer_stateless/decode.py:2]: Oups my bad. Fixed

File: sherpa/csrc/rnnt_emformer_model.cc
Comment (csukuangfj) [inline sherpa/csrc/rnnt_emformer_model.cc:56]: For the following CI error
https://github.com/k2-fsa/sherpa/runs/7444603197?check_suite_focus=true

You can check the style of the C++ code locally by
```bash
./scripts/check_style_cpplint.sh 2
```

Comment (ezerhouni) [reply sherpa/csrc/rnnt_emformer_model.cc:56]: Yup, already working on it ;)

**Comment by ezerhouni** [comment]
Should be all good @csukuangfj !

**Review by csukuangfj — APPROVED** [review]

**Comment by csukuangfj** [comment]
> Should be all good @csukuangfj !

Thanks! Let us wait for the CI.  Will merge after all tests pass. 

Thanks again!

**Pull request merged**
